### PR TITLE
add duckdb data source

### PR DESCRIPTION
**Pull request by geertjan-garvis**
first contribution, so creating as draft first. let me know if I missed anything

### CODE REVIEW

File: soda/duckdb/soda/data_sources/duckdb_data_source.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        "double",
+        "real",
+    ]
+    TEXT_TYPES_FOR_PROFILING = ["char", "varchar"]
+
+    def __init__(self, logs: Logs, data_source_name: str, data_source_properties: dict):
+        super().__init__(logs, data_source_name, data_source_properties)
+        self.path = data_source_properties.get("path")
+        self.read_only = data_source_properties.get("read_only", False)
+        self.duckdb_connection = data_source_properties.get("duckdb_connection")
```
Comment (JorgeGarciaIrazabal) [inline soda/duckdb/soda/data_sources/duckdb_data_source.py:99]: Not a blocker, but could you add a parameter for [configuration](https://duckdb.org/docs/sql/configuration.html)?
Users might want to have specific configuration when running soda checks.

**Comment by vijaykiran** [comment]
@geertjan-garvis Not sure what went wrong in the git flow, I'll pull your commits separately and rebase to create a linear history from a different branch.

**Comment by vijaykiran** [comment]
closing this in favour of - https://github.com/sodadata/soda-core/pull/1709

File: soda/duckdb/soda/data_sources/duckdb_data_source.py (hunk 1 above)
Comment (vijaykiran) [reply soda/duckdb/soda/data_sources/duckdb_data_source.py:99]: @JorgeGarciaIrazabal  Can you please open an issue for it - we can tackle as a separate PR

**Pull request closed without merging**